File: benchmarks/pandas/bench_dataframe_round.py
import pandas as pd, time, json
N = 100_000
cols = 5
data = {f"col{c}": [(i % 100) * 1.5 for i in range(N)] for c in range(cols)}
df = pd.DataFrame(data)
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    df.round(2)
t0 = time.perf_counter()
for _ in range(ITERS):
    df.round(2)
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "dataframe_round", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))

File: benchmarks/tsb/bench_dataframe_round.ts
import { DataFrame } from "tsb";
const N = 100_000;
const cols = 5;
const data: Record<string, number[]> = {};
for (let c = 0; c < cols; c++) {
  data[`col${c}`] = Array.from({ length: N }, (_, i) => (i % 100) * 1.5);
}
const df = new DataFrame(data);
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) df.round(2);
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) df.round(2);
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "dataframe_round", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));
